src/App.js

```diff
@@ -12,138 +12,101 @@ import People from './components/People';
 import NicetyDisplay from './components/NicetyDisplay';
 import Admin from './components/Admin';
 
-
 if (store.get("saved") === null) {
     store.set("saved", true);
 }
 
-const App = React.createClass({
-    loadPeopleFromServer: function(callback) {
-        $.ajax({
-            url: this.props.people_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.people, status, err.toString());
-            }
-        });
-    },
-    loadNicetiesFromMe: function(callback) {
-        $.ajax({
-            url: this.props.from_me_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.fromMe, status, err.toString());
-            }
-        });
-    },
-    loadNicetiesForMe: function(callback) {
-        $.ajax({
-            url: this.props.for_me_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.niceties, status, err.toString());
-            }
-        });
-    },
-    loadSelfInfo: function(callback) {
-        $.ajax({
-            url: this.props.self_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.niceties, status, err.toString());
-            }
-        });
-    },
-    getInitialState: function() {
-        return {
-                fromMe: [],
-                people: [],
-                niceties: [],
-                currentview: "write-niceties",
-                selfInfo: [],
-        };
-    },
-    componentDidMount: function() {
-      this.loadNicetiesFromMe((data) => {
-        this.setState({fromMe: data})
-      })
-      this.loadPeopleFromServer((data) => {
-        this.setState({people: data})
-      })
-      this.loadNicetiesForMe((data) => {
-        this.setState({niceties: data})
+class App extends React.Component {
+    constructor(props) {
+      super(props)
+      this.state = {
+        fromMe: [],
+        people: [],
+        niceties: [],
+        currentview: "write-niceties",
+        selfInfo: []
+      }
+    }
+
+    loadData = (api, dataVar) => {
+      fetch(api, {
+        headers: {'Content-Type': "application/json"},
+        cache: 'no-cache'
```

Comment:
This is a faithful translation of the jQuery code, and I appreciate it! I hadn't noticed before that we had this no-cache directive, and it makes me think the API is not correctly communicating its cacheability. Definitely nothing to change here, but maybe something to investigate later!